tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -70,66 +83,100 @@ func testDestroyWindowsInstance(t *testing.T) {
+	if !ok {
+		return fmt.Errorf("error asserting cloudprovider to awsProvider")
+	}
 	awsProvider = provider
+	return nil
 }
 
 // setupWindowsInstanceWithResources creates a Windows instance and updates global information for infraID,
 // createdInstanceID, and createdSgID. All information updates are required to be successful or instance will be
 // teared down.
-func setupWindowsInstanceWithResources(t *testing.T) {
+func setupWindowsInstanceWithResources() error {
 	var err error
+	// Create the instance
 	credentials, err = awsProvider.CreateWindowsVM()
 	if err != nil {
-		tearDownInstance(t, "error creating Windows instance", err)
+		return fmt.Errorf("error creating Windows instance: %s", err)
+	}
+
+	// Ensure we have the login info for the instance
+	if credentials == nil {
+		return fmt.Errorf("returned credentials empty")
+	}
+	if credentials.GetPassword() == "" {
+		return fmt.Errorf("returned password empty")
+	}
+	if credentials.GetInstanceId() == "" {
+		return fmt.Errorf("returned instance id empty")
 	}
-	require.NoError(t, err, "failed to create the windows instance")
 
 	infraID, err = awsProvider.GetInfraID()
 	if err != nil {
-		tearDownInstance(t, "error while getting infrastructure ID for the OpenShift cluster", err)
+		return fmt.Errorf("error while getting infrastructure ID for the OpenShift cluster: %s", err)
 	}
-	require.NotEmpty(t, credentials, "Credentials returned are empty")
-	require.NotEmpty(t, credentials.GetPassword(), "Expected some password but got empty string")
-	require.NotEmpty(t, credentials.GetInstanceId(), "Expected instanceID to be present but got empty string")
 
 	// Check instance and security group information in windows-node-installer.json.
 	info, err := resource.ReadInstallerInfo(artifactDir + "/" + "windows-node-installer.json")
 	if err != nil {
-		tearDownInstance(t, "error reading from windows-node-installer.json file", err)
+		return fmt.Errorf("error reading from windows-node-installer.json file: %s", err)
 	}
 
-	if len(info.SecurityGroupIDs) > 0 && info.SecurityGroupIDs[0] != "" {
-		createdSgID = info.SecurityGroupIDs[0]
-	} else {
-		tearDownInstance(t, "security group ID information is empty", err)
+	// Set security group value
+	if len(info.SecurityGroupIDs) != 1 {
+		return fmt.Errorf("expected one security group but found %d", len(info.SecurityGroupIDs))
 	}
+	if info.SecurityGroupIDs[0] == "" {
+		return fmt.Errorf("found empty security group")
+	}
+	createdSgID = info.SecurityGroupIDs[0]
 
-	if len(info.SecurityGroupIDs) > 0 && info.SecurityGroupIDs[0] != "" {
-		createdInstanceID = info.InstanceIDs[0]
-	} else {
-		tearDownInstance(t, "instance ID information is empty", err)
+	// Set instanceID value
+	if len(info.InstanceIDs) != 1 {
+		return fmt.Errorf("expected one instance but found %d", len(info.InstanceIDs))
+	}
+	if info.InstanceIDs[0] == "" {
+		return fmt.Errorf("found empty instance id value")
 	}
+	createdInstanceID = info.InstanceIDs[0]
```

Comment:
We  should get rid of `createdInstanceId` given that it is redundant as the same info is in the `credentials`. Just calling it out here so that everyone is aware and we don't do more with that variable.